///////////////////////////////////////////////////
//         Define Protobuf Syntax Version        //         
///////////////////////////////////////////////////
syntax = "proto3";

///////////////////////////////////////////////////
//              Define Package Name              //         
///////////////////////////////////////////////////
package gpio;

///////////////////////////////////////////////////
//               Define RPC Service              //         
///////////////////////////////////////////////////
service GPIO {
    // Turn On PYNQ SDN 1 I/O
    rpc TurnOnPYNQSDN1IO(TurnOnPYNQSDN1IOPacket) returns (TurnOnPYNQSDN1IOResponse) {}
    // Turn Off PYNQ SDN 1 I/O
    rpc TurnOffPYNQSDN1IO(TurnOffPYNQSDN1IOPacket) returns (TurnOffPYNQSDN1IOResponse) {}
    // Turn On PYNQ SDN 2 I/O
    rpc TurnOnPYNQSDN2IO(TurnOnPYNQSDN2IOPacket) returns (TurnOnPYNQSDN2IOResponse) {}
    // Turn Off PYNQ SDN 2 I/O
    rpc TurnOffPYNQSDN2IO(TurnOffPYNQSDN2IOPacket) returns (TurnOffPYNQSDN2IOResponse) {}
    // Turn On PYNQ SDN 3 I/O
    rpc TurnOnPYNQSDN3IO(TurnOnPYNQSDN3IOPacket) returns (TurnOnPYNQSDN3IOResponse) {}
    // Turn Off PYNQ SDN 3 I/O
    rpc TurnOffPYNQSDN3IO(TurnOffPYNQSDN3IOPacket) returns (TurnOffPYNQSDN3IOResponse) {}
    // Reset ADC Start
    rpc ResetADCStart(ResetADCStartPacket) returns (ResetADCStartResponse) {}
    // Start ADC
    rpc StartADC(StartADCPacket) returns (StartADCResponse) {}
}

///////////////////////////////////////////////////
//              Define Message Type              //         
///////////////////////////////////////////////////
// Turn On PYNQ SDN 1 I/O Message
message TurnOnPYNQSDN1IOPacket {}

// Turn On PYNQ SDN 1 I/O Response
message TurnOnPYNQSDN1IOResponse {
    // Define the fields
    string operation_name = 1;
    string start_time = 2;
    string end_time = 3;
    bool success = 4;
}

// Turn Off PYNQ SDN 1 I/O Message
message TurnOffPYNQSDN1IOPacket {}

// Turn Off PYNQ SDN 1 I/O Response
message TurnOffPYNQSDN1IOResponse {
    // Define the fields
    string operation_name = 1;
    string start_time = 2;
    string end_time = 3;
    bool success = 4;
}

// Turn On PYNQ SDN 2 I/O Message
message TurnOnPYNQSDN2IOPacket {}

// Turn On PYNQ SDN 2 I/O Response
message TurnOnPYNQSDN2IOResponse {
    // Define the fields
    string operation_name = 1;
    string start_time = 2;
    string end_time = 3;
    bool success = 4;
}

// Turn Off PYNQ SDN 2 I/O Message
message TurnOffPYNQSDN2IOPacket {}

// Turn Off PYNQ SDN 2 I/O Response
message TurnOffPYNQSDN2IOResponse {
    // Define the fields
    string operation_name = 1;
    string start_time = 2;
    string end_time = 3;
    bool success = 4;
}

// Turn On PYNQ SDN 3 I/O Message
message TurnOnPYNQSDN3IOPacket {}

// Turn On PYNQ SDN 3 I/O Response
message TurnOnPYNQSDN3IOResponse {
    // Define the fields
    string operation_name = 1;
    string start_time = 2;
    string end_time = 3;
    bool success = 4;
}

// Turn Off PYNQ SDN 3 I/O Message
message TurnOffPYNQSDN3IOPacket {}

// Turn Off PYNQ SDN 3 I/O Response
message TurnOffPYNQSDN3IOResponse {
    // Define the fields
    string operation_name = 1;
    string start_time = 2;
    string end_time = 3;
    bool success = 4;
}

// Reset ADC Start Message
message ResetADCStartPacket {}

// Reset ADC Start Response
message ResetADCStartResponse {
    // Define the fields
    string operation_name = 1;
    string start_time = 2;
    string end_time = 3;
    bool success = 4;
}

// Start ADC Message
message StartADCPacket {
    // Define the fields
    float time_delay_1 = 1;
    float time_delay_2 = 2;
}

// Start ADC Response
message StartADCResponse {
    // Define the fields
    string operation_name = 1;
    string start_time = 2;
    string end_time = 3;
    bool success = 4;
}